common/include/pcl/experimental/executor/inline_executor.hpp

```diff
@@ -42,18 +48,30 @@ struct inline_executor {
     return !operator==(lhs, rhs);
   }
 
-  template <typename F>
+  /**
+   * \brief Launches a single execution agent which invokes the callable
+   *
+   * \param f a callable
+   */
+  template <typename Function>
   void
-  execute(F&& f) const
+  execute(Function&& f) const
   {
     f();
   }
 
-  template <typename F, typename... Args>
+  /**
+   * \brief Eagerly launches execution agents in bulk and which invoke
+   * the callable with the associated agent's index.
+   *
+   * \param f a callable
```

Comment:
Should we note and assert the type of callable?

Reply:
Will open up an issue post GSoC